pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/ContextImpl.java

```diff
@@ -91,6 +91,7 @@ public void update(double value) {
         }
     }
 
+    private ConcurrentMap<String, AccumulatedMetricDatum> currentAccumulatedMetrics;
```

Comment:
I believe you can use the same var for both accumulatedMetrics and currentAccumulatedMetrics.

Reply:
umm.. I didn't get it. `currentAccumulatedMetrics` captures current ongoing metrics and `accumulatedMetrics` will keep the sample which can be returned when getMetrics api is called.